app/views/RegisterView/index.tsx

```diff
@@ -106,7 +112,13 @@ const RegisterView = ({ navigation, route }: IProps) => {
 	const onSubmit = async (data: any) => {
 		logEvent(events.REGISTER_DEFAULT_SIGN_UP);
 
-		const { name, email, password, username } = data;
+		const { name, email, password, username, confirmPassword } = data;
+
+		if (password !== confirmPassword) {
+			setError('confirmPassword', { message: I18n.t('Passwords_do_not_match'), type: 'validate' });
+			AccessibilityInfo.announceForAccessibility(I18n.t('Passwords_do_not_match'));
+			return;
+		}
```

Comment:
Can't you do something like this?

https://stackoverflow.com/a/63187464/5447468
https://github.com/jquense/yup/issues/682